client/src/components/Donation/donation-modal.tsx

```diff
@@ -62,13 +66,22 @@ function DonateModal({
     if (isLocationSuperBlock(location)) {
       goToAnchor('claim-cert-block');
     }
+
+    if (isA11yFeatureEnabled) {
+      if (!canClose) {
+        return;
+      }
+
+      closeDonationModal();
+    }
```

Comment:
Wouldn't the following be more straightforward. 
```suggestion
    if (isA11yFeatureEnabled && canClose ) {
              closeDonationModal();
    }
```